src/entities/TransactionQueue.ts

```diff
@@ -19,39 +19,89 @@ enum Events {
   TransactionStatusChange = 'TransactionStatusChange',
 }
 
+/**
+ * Class to manage procedural transaction queues
+ */
 export class TransactionQueue<Args extends any = any, ReturnType extends any = void> extends Entity<
   void
 > {
+  /**
+   * Generate an ID for a transaction
+   */
   public static generateId() {
     return serialize('transaction', {
       random: v4(),
     });
   }
 
+  /**
+   * Type of entity
+   */
   public readonly entityType: string = 'transactionQueue';
 
+  /**
+   * Type of procedure being run
+   */
   public procedureType: ProcedureType;
 
+  /**
+   * Generated transaction queue unique identifier
+   */
   public uid: string;
 
+  /**
+   * Array of poly transactions
+   */
   public transactions: PolyTransaction[];
 
+  /**
+   * Status of the transaction queue
+   */
   public status: TransactionQueueStatus = TransactionQueueStatus.Idle;
 
+  /**
+   * Arguments provided to the transaction queue
+   */
   public args: Args;
 
+  /**
+   * Optional error information
+   */
   public error?: Error;
 
+  /**
+   * Associated Fees for the transaction
+   */
   public fees: Fees;
 
+  /**
+   * @hidden
+   */
   private promise: Promise<ReturnType>;
 
+  /**
+   * @hidden
+   */
   private queue: PolyTransaction[] = [];
 
+  /**
+   * @hidden
+   */
   private returnValue: MaybeResolver<ReturnType>;
 
+  /**
+   * @hidden
+   */
   private emitter: EventEmitter;
 
+  /**
+   * Create a transaction queue
+   * @param transactions transaction specifications
+   * @param fees required fees to make a transaction
+   * @param returnValue value returned by a resolver attached to the transaction
```

Comment:
value that will be returned by the queue after it is run. It can be a Post Transaction Resolver